From f1e5c3bb0d74fbf0caf17c5f03cfd8e2fcf08268 Mon Sep 17 00:00:00 2001
From: Michal Rostecki <mrostecki@opensuse.org>
Date: Tue, 14 Jan 2020 03:11:27 +0100
Subject: [PATCH] logger: Use spdlog::memory_buf_t instead of
 fmt::memory_buffer

The newest spdlog has its own type spdlog::memory_buf_t, which is equal
to the type fmt::basic_memory_buffer<char, 250>.

Risk Level: Low
Testing: N/A
Docs Changes: N/A

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>
---
 source/common/common/logger.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/envoy/source/common/common/logger.cc b/envoy/source/common/common/logger.cc
index 5b6ea7bee..ef5bcafe1 100644
--- a/envoy/source/common/common/logger.cc
+++ b/envoy/source/common/common/logger.cc
@@ -62,7 +62,7 @@ void DelegatingLogSink::log(const spdlog::details::log_msg& msg) {
 
   // This memory buffer must exist in the scope of the entire function,
   // otherwise the string_view will refer to memory that is already free.
-  fmt::memory_buffer formatted;
+  spdlog::memory_buf_t formatted;
   if (formatter_) {
     formatter_->format(msg, formatted);
     msg_view = absl::string_view(formatted.data(), formatted.size());
-- 
2.16.4

